### PR TITLE
fix(eventpipeline): Nonstring transactions cause an error during event ingestion 

### PR DESCRIPTION
**Pull request by lauryndbrown**

### CODE REVIEW

File: src/sentry/event_manager.py [hunk 1]
```diff
@@ -510,6 +509,9 @@ def save(self, project, raw=False):
         else:
             transaction_name = culprit
 
+        if not isinstance(culprit, six.string_types):
```
Comment (mattrobenolt) [inline src/sentry/event_manager.py]: There’s no need to check the type, `force_text` will just noop if the input is already unicode.

**Comment by mattrobenolt** [comment]
SHould we maybe just do this inside of `generate_culprit` instead? Seems odd to me to do this coercion inside of `save`.

**Comment by lauryndbrown** [comment]
@mattrobenolt that's a good question, but not in this case. If you take a look at what I'm doing in the test, the transaction value is actually the value that is put into the culprit. And it is bypassing generate culprit. I realized that was what was causing the issue in the error we were getting. That said, if generate_culprit also generates a nonstring value, this will take care of that as well. 

https://github.com/getsentry/sentry/blob/16a02f86bd3949abf12974c6a5878c9421fef47a/src/sentry/event_manager.py#L487

https://github.com/getsentry/sentry/blob/master/src/sentry/event_manager.py#L505:L511

**Comment by mattrobenolt** [comment]
Ah, great observation. Makes sense.

**Review by mattrobenolt — APPROVED** [review]
Aside from my nit about not needing to check the type before passing through `force_text`, looks great! 🦄

**Pull request merged**